lib/pluginmanager/generate.rb

```diff
@@ -35,6 +32,10 @@ def execute
 
   private
 
+  def validate_params
+    raise(ArgumentError, "should be one of: input, output or filter") unless TYPES.include?(type)
```

Comment:
agreed, just tried to split the PR as I was not expecting the codec to go in so fast 👍